src/sliding_windows.rs

```diff
@@ -1,15 +1,17 @@
+use std::cell::{Cell, UnsafeCell};
 use std::mem;
-use std::ops::{
-    Deref, DerefMut};
+use std::ops::{Deref, DerefMut};
+
+use std::marker::PhantomData;
+use std::fmt;
 
 // This stores the current window
-#[allow(raw_pointer_derive)]
-#[derive(Debug)]
 pub struct SlidingWindowStorage<T> {
     window_size: usize,
-    uniquely_owned: bool,
-    data: Vec<T>,
-    _no_send_sync_marker: *const u8 //TODO remove in favour of negative send and sync impls once they are stable
+    /// acts as a refcount
+    uniquely_owned: Cell<bool>,
+    data: UnsafeCell<Vec<T>>,
+    _no_send_sync_marker: PhantomData<*const ()>,
```

Comment:
I forgot that this wasn't needed -- Cell or UnsafeCell is enough. This whole commit is in a “WIP” quality anyway.